tests/lib/fortigate_test.py

```diff
@@ -0,0 +1,390 @@
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unittest for fortigate policy rendering module."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import string
+import re
+import unittest
+
+from capirca.lib import fortigate
+from capirca.lib import nacaddr
+from capirca.lib import naming
+from capirca.lib import policy
+import mock
+
+
+GOOD_HEADER = """
+header {
+  comment:: "this is a test acl"  
+  target:: fortigate from-id 2
+}
+"""
+
+BAD_HEADER = """
+header {
+  comment:: "this is a test acl"
+  target:: fortigate edge-filter
+}
+"""
+
+TERM_TEMPLATE = """
+term good-term-2 {{
+  source-interface:: {src_interface}
+  destination-interface:: {dest_interface}
+  protocol:: {protocol}
+  destination-address:: {dest_addr}
+  destination-port:: {dest_port}
+  source-address:: {src_addr}
+  source-port:: {src_port}
+  action:: {action}
+  logging:: {logging}
+}}
+"""
+
+
+class CustomFormatter(string.Formatter):
+  DEFAULT_VALUES = {
+      'src_interface': 'wan1',
+      'dest_interface': 'wan2',
+      'protocol': 'tcp',
+      'src_addr': 'SOME_HOST',
+      'dest_addr': 'SOME_HOST',
+      'src_port': 'HTTP',
+      'dest_port': 'HTTP',
+      'action': 'accept',
+      'logging': 'true'
+  }
+
+  def format(*args, **kwargs):
+    if 'remove_fields' in kwargs:
+      args = list(args)
+      for field in kwargs['remove_fields']:
+        remove_regex = '.*' + field + '.*'
+        args[1] = re.sub(remove_regex, '', args[1])
+
+      return string.Formatter.format(*args, **kwargs)
+    return string.Formatter.format(*args, **kwargs)
+
+  def get_value(self, key, args, kwds):
+    try:
+      return kwds[key]
+    except KeyError:
+      return self.DEFAULT_VALUES[key]
+
+
+EXP_INFO = 2
+
+
+class FortigateTest(unittest.TestCase):
+  def setUp(self):
```

Comment:
one new line should be between new definitions.
http://google.github.io/styleguide/pyguide.html#35-blank-lines

also no call to super in the setUp, this should be done.

Reply:
the super call is not needed in this case, [same here](https://github.com/Ali-aqrabawi/capirca/blob/cf91480d58e4a96d9f5e45f2609755d2c62a6c9c/tests/lib/cisco_test.py#L412) and for other `test modules`

Reply:
New line is still required.

Reply:
Closed without resolution